### PR TITLE
Update examples (fixes #21)

### LINKED ISSUE #21: Update all examples to include images
Add images to all examples and link the used `gantry.def.yml` and `gantry.env.yml` files.

### PR DESCRIPTION
**Pull request by lehmann-4178656ch**
* Updates all examples to fix #21

* Adds README.md to examples folder explaining the shapes and how the images can are generated.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Examples
+
+![types.svg](./types.svg)
+
+Gantry supports `services` (oval shapes) and `steps` (box shapes).
```
Comment (niklas88) [inline examples/README.md]: I think this would be a great place to add a sentence or two about the difference between them

Comment (lehmann-4178656ch) [reply examples/README.md]: I've added some explanation to this.

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Services are docker containers executed like `docker-compose` is used, they
```
Comment (niklas88) [inline examples/README.md]: This is already a good description (except for a grammar error) of the *how* but doesn't really say anything about the *why*. Here's my take that takes your facts for the *how* and mixes in the *why*.

"Services define docker containers which provide a continued service to other task in and outside the pipeline. They directly resemble the service concept from `docker-compose` and once started run concurrently to the rest of the pipeline including anything depending on the service.

Steps on the other hand run to completion and only then are their dependents executed. Note however, that independent steps are executed concurrently with each other. Steps are often used for tasks that produce a result that is needed by their dependents such as a download, creation of a database index or training of a machine learning model."

Also I think we should add this part to the main README

File: examples/README.md (hunk 1 above)
Comment (lehmann-4178656ch) [reply examples/README.md]: Moved to the main readme as suggested.

File: examples/README.md (hunk 2 above)
Comment (lehmann-4178656ch) [reply examples/README.md]: Thank you for the text, I've added it to the main README now

File: examples/qlever_e2e/README.md [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+
+![pipeline.svg](./pipeline.svg)
+
 This example runs the End-to-End tests of
 [QLever](https://github.com/ad-freiburg/QLever). It's an adaptation of the
 [shell script](https://github.com/ad-freiburg/QLever/blob/b728ddd0f3ac3e4c9d1f0e92426f5b592a4f141f/e2e/e2e.sh)
 using multiple steps and starting a service (qlever).
 
-The structure of the example is visualized in [pipeline.svg](./pipeline.svg).
+The pipeline downloads and prepares the data create an index for qlever. After
```
Comment (niklas88) [inline examples/qlever_e2e/README.md]: missing "to"

File: README.md [hunk 4]
```diff
@@ -1,61 +1,95 @@
...
 
 Gantry is a pipeline management tool using containers for all relevant steps.
 It supports a basic `docker-compose` subset allowing `docker-compose` like
 deployments with [wharfer](https://github.com/ad-freiburg/wharfer). If `wharfer`
 is not installed `docker` will be used directly.
 
+## Services and Steps
+
+Services define docker containers which provide a continued service to other
+task in and outside of the pipeline. They directly resemble the service
```
Comment (niklas88) [inline README.md]: tasks

File: README.md [hunk 5]
```diff
@@ -1,61 +1,95 @@
...
+concept from docker-compose and once started run concurrently to the rest of
```
Comment (niklas88) [inline README.md]: `docker-compose` to match above use

File: README.md [hunk 6]
```diff
@@ -1,61 +1,95 @@
...
 
-Finally use the GitHub Releases mechanism to release a new version
+### From source
 
-## Installing from source
 To install gantry into the users ~/go/bin path it is enough to just run
 
     go install -ldflags="-X github.com/ad-freiburg/gantry.Version=$(git describe --always --long --dirty)" ./...
 
+To install gantry global copy it from ~/go/bin path or use
```
Comment (niklas88) [inline README.md]: globally

File: README.md (hunk 4 above)
Comment (lehmann-4178656ch) [reply README.md]: Changed

File: README.md (hunk 5 above)
Comment (lehmann-4178656ch) [reply README.md]: Changed

File: README.md (hunk 6 above)
Comment (lehmann-4178656ch) [reply README.md]: Thanks, that's the word I thought I used.

**Pull request merged**